projects/oit_demo/OITDemoApplication.cpp

```diff
@@ -289,56 +292,79 @@ void OITDemoApp::SetupCommon()
     }
 }
 
-void OITDemoApp::AddSupportedAlgorithm(const char* name, Algorithm algorithm)
+OITDemoApp::Algorithm OITDemoApp::GetSelectedAlgorithm() const
 {
-    mSupportedAlgorithmNames.push_back(name);
-    mSupportedAlgorithmIds.push_back(algorithm);
-    PPX_ASSERT_MSG(mSupportedAlgorithmNames.size() == mSupportedAlgorithmIds.size(), "supported algorithm data is out-of-sync");
+    return mSupportedAlgorithmIds[mGuiParameters.algorithmDataIndex];
 }
 
-OITDemoApp::Algorithm OITDemoApp::GetSelectedAlgorithm() const
+grfx::MeshPtr OITDemoApp::GetTransparentMesh() const
 {
-    return mSupportedAlgorithmIds[mGuiParameters.algorithmDataIndex];
+    return mTransparentMeshes[mGuiParameters.mesh.type];
 }
 
 void OITDemoApp::FillSupportedAlgorithmData()
 {
-    AddSupportedAlgorithm("Unsorted over", ALGORITHM_UNSORTED_OVER);
-    AddSupportedAlgorithm("Weighted sum", ALGORITHM_WEIGHTED_SUM);
+    const auto addSupportedAlgorithm = [this](const char* name, Algorithm algorithm) {
+        mSupportedAlgorithmNames.push_back(name);
+        mSupportedAlgorithmIds.push_back(algorithm);
+        PPX_ASSERT_MSG(mSupportedAlgorithmNames.size() == mSupportedAlgorithmIds.size(), "supported algorithm data is out-of-sync");
```

Comment:
I use the static assert to make sure the array are in sync size-wise. As long as the function is used, it should always be the case, but who knows if I break that at some point by mistake.
As you guessed, they are separate array so that I can use  mSupportedAlgorithmNames with ImGui.
As for the dynamic aspect, it is because some of algorithms require particular hardware capabilities, so not all algorithms are always available. I will actually extend that in a subsequent PR, as I realized I always call the Setup*** function even when the hardware capabilities is not present (and that would trigger the validation error, even though the whole point of this was to avoid validation errors).

Reply:
Ok I see, thanks.